### ISSUE TITLE
[R] `predict` doesn't return probabilities for `multi:softmax`

### ISSUE THREAD

**Issue by david-cortes** [dmlc/xgboost#7935]
When using objective `multi:softmax`, calls to predict will always return the predicted class instead of the probabilities:
```r
library(xgboost)
data(iris)
X <- as.matrix(iris[, -5])
y <- as.numeric(iris$Species) - 1
bst_multi <- xgboost(data = X, label = y, objective = "multi:softmax",
                     nrounds = 5L, verbose = FALSE, num_class = 3L)
pred <- predict(bst_multi, X, strict_shape=TRUE)
dim(pred)
```
```
[1]   1 150
```

According to the docs, this should produce an output with shape `(3, 150)`.

**Comment by hcho3** [comment]
Using multi:softmax objective will produce class predictions. For now, please use multi:softprob.

I would welcome your suggestion for improving the behavior. I can think of following options:

1. Forbid multi:softmax objective in the R interface. Always use multi:softprob.
2. Make multi:softmax behave like multi:softprob at prediction time.

**Comment by david-cortes** [comment]
I think the 3rd one would be the most ideal.

**Comment by trivialfis** [comment]
I think those are the same objective and the only difference is the softmax collapses probability into class index at prediction time.
